### PR TITLE
AB#32820: Honeypot Registration forms

### PR DESCRIPTION
**Pull request by shakisparki**
## Overview

Adds a hidden honey pot field.
If set to true during registration, User is added to the block list.

### To Do

- [x] Add bool Honey pot field to model
- [x] Add functionality for Biobank Registration
- [x] Add checkbox input to shared partial view 
- [x] Add functionality for Network Registration
- [x] Add JavaScript for checkbox

### Note
PR branches off of `blockallowEmailDomain` branch work in progress by @bradthompson995.
The PR from  `blockallowEmailDomain` should be merged before this.

### CODE REVIEW

File: src/Directory/Biobanks.Web/Controllers/ADACController.cs [hunk 1]
```diff
@@ -1669,6 +1669,28 @@ public ActionResult HomepageConfigPreview()
...
+                {
+                    Id = x.Id,
+                    RuleType = x.RuleType,
+                    Value = x.Value,
+                    Source = x.Source,
+                    DateModified = x.DateModified
+                }));
+         
+        }
+
```
Comment (Flemingjp) [inline src/Directory/Biobanks.Web/Controllers/ADACController.cs:1691]: Can we remove the additional blank lines?

File: src/Directory/Biobanks.Web/Controllers/RegisterController.cs [hunk 2]
```diff
@@ -62,6 +63,42 @@ public ActionResult AdacBiobank()
             });
         }
 
+        internal async Task<bool> RegistrationHoneypotTrap(RegisterEntityModel model)
+        {
```
Comment (Flemingjp) [inline src/Directory/Biobanks.Web/Controllers/RegisterController.cs:67]: Is `internal` required here?

File: src/Directory/Biobanks.Web/ApiControllers/RegistrationDomainRuleController.cs [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+        {
+            _biobankReadService = biobankReadService;
+            _biobankWriteService = biobankWriteService;
+        }
+
+        [HttpGet]
+        [Route("")]
+        public async Task<IList> Get()
+        {
+            var model = (await _biobankReadService.ListRegistrationDomainRulesAsync())
```
Comment (Flemingjp) [inline src/Directory/Biobanks.Web/ApiControllers/RegistrationDomainRuleController.cs:31]: Might as well have it as a single return statement here.

File: src/Directory/Biobanks.Web/ApiControllers/RegistrationDomainRuleController.cs [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+        public RegistrationDomainRuleController(IBiobankReadService biobankReadService,
+                                          IBiobankWriteService biobankWriteService)
+        {
+            _biobankReadService = biobankReadService;
+            _biobankWriteService = biobankWriteService;
+        }
+
+        [HttpGet]
+        [Route("")]
+        public async Task<IList> Get()
```
Comment (Flemingjp) [inline src/Directory/Biobanks.Web/ApiControllers/RegistrationDomainRuleController.cs:29]: Is `IList` used over `IEnumerable` to ensure Json payload here?

File: src/Directory/Biobanks.Web/ApiControllers/RegistrationDomainRuleController.cs [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+
+            await _biobankWriteService.AddRegistrationDomainRuleAsync(type);
+            await _biobankWriteService.UpdateRegistrationDomainRuleAsync(type); // Ensure sortOrder is correct
+
+            // Success response
+            return Json(new
+            {
+                success = true,
+                name = model.Value,
+            });
```
Comment (Flemingjp) [inline src/Directory/Biobanks.Web/ApiControllers/RegistrationDomainRuleController.cs:80]: Do we need to have a success variable here - a HTTP 200 OK should be sufficient? @beforan?

**Review by Flemingjp — CHANGES_REQUESTED** [review]

File: src/Directory/Biobanks.Web/Controllers/ADACController.cs (hunk 1 above)
Comment (shakisparki) [reply src/Directory/Biobanks.Web/Controllers/ADACController.cs:1691]: These snippets are from Bradley's PR. Code for this PR should be more visible once he's merged his work.

File: src/Directory/Biobanks.Web/ApiControllers/RegistrationDomainRuleController.cs (hunk 3 above)
Comment (shakisparki) [reply src/Directory/Biobanks.Web/ApiControllers/RegistrationDomainRuleController.cs:31]: These snippets are from Bradley's PR

File: src/Directory/Biobanks.Web/ApiControllers/RegistrationDomainRuleController.cs (hunk 4 above)
Comment (shakisparki) [reply src/Directory/Biobanks.Web/ApiControllers/RegistrationDomainRuleController.cs:29]: These snippets are from Bradley's PR

File: src/Directory/Biobanks.Web/ApiControllers/RegistrationDomainRuleController.cs (hunk 5 above)
Comment (shakisparki) [reply src/Directory/Biobanks.Web/ApiControllers/RegistrationDomainRuleController.cs:80]: These snippets are from Bradley's PR

**Review by Flemingjp — APPROVED** [review]

**Review by blargly — APPROVED** [review]

**Pull request merged**